docs/how_tos/storage_backends.md

```diff
@@ -0,0 +1,359 @@
+
+**Characteristics:**
+
+- Uses HDF5 file format, popular in scientific computing
+- Supports hierarchical organization of data
+- Good compression capabilities
+- Compatibility with PSY
+
+```python
+system = System(time_series_storage_type=TimeSeriesStorageType.HDF5)
+```
+
+```{note}
+HDF5 storage is currently available in the development version only.
+```
+
+## Working with Time Series Data
+
+Regardless of the backend you choose, the API for adding, retrieving, and using time series data remains the same:
+
+```python
+from datetime import datetime, timedelta
+import numpy as np
+from infrasys.time_series_models import SingleTimeSeries
+from tests.models.simple_system import SimpleSystem, SimpleGenerator, SimpleBus
+
+# Create a system with your chosen backend
+system = SimpleSystem(time_series_storage_type=TimeSeriesStorageType.ARROW)
+
+# Add components
+bus = SimpleBus(name="test-bus", voltage=1.1)
+generator = SimpleGenerator(name="gen", active_power=1.0, rating=1.0, bus=bus, available=True)
+system.add_components(bus, generator)
+
+# Create and add time series data
+length = 24
+initial_time = datetime(year=2020, month=1, day=1)
+resolution = timedelta(hours=1)
+data = np.random.rand(length)
+
+# Create a time series
+time_series = SingleTimeSeries(
+    data=data,
+    name="active_power",
+    resolution=resolution,
+    initial_timestamp=initial_time
+)
+
+# Add the time series to a component
+system.add_time_series(time_series, generator, scenario="baseline")
+
+# Retrieve the time series later
+retrieved_ts = system.get_time_series(
+    generator,
+    name="active_power",
+    scenario="baseline"
+)
+```
+
+## Read-Only Mode
+
+For any storage backend, you can set it to read-only mode, which is useful when you're working with existing data that shouldn't be modified:
+
+```python
+system = System(time_series_read_only=True)
+```
+
+```{warning}
+In read-only mode, attempts to add or modify time series data will raise exceptions.
+```
+
+## Serializing and Deserializing a System
+
+When saving a system to disk, all the time series data will be properly serialized regardless of the backend used:
+
+```python
+from pathlib import Path
+
+# Save the entire system (including time series data)
+output_dir = Path("my_system_data")
+system.to_json(output_dir)
+
+# To load the system back
+loaded_system = SimpleSystem.from_json(output_dir)
+```
+
+```{note}
+The storage backend information is preserved when saving and loading a system.
+```
+
+## Performance Considerations
+
+Each storage backend offers different trade-offs in terms of performance:
+
+- **Memory Usage**: In-memory storage keeps all data in RAM, which can be a limitation for large datasets
+- **Disk Space**: Arrow, Chronify, and HDF5 storage use disk space, with different compression characteristics
+- **Access Speed**: In-memory is fastest, followed by Arrow, then Chronify and HDF5 (depending on the specific operation)
```

Comment:
I would expect HDF5 to be close to Arrow and much faster than Chronfiy. Is that not true?

Reply:
Correct. I modified to show tha arrow/HDF5 has similar speeds